module/lua/ldo.c

```diff
@@ -29,6 +29,18 @@
 
 
 
+/* Return the number of bytes available on the stack. */
+#if defined (_KERNEL) && defined(__linux__)
```

Comment:
It looks to me like this issue isn't specific to Linux, FreeBSD's default stack size is also 16K (KSTACK_PAGES=4).

Reply:
Yes, the same LUA test program instantly panics FreeBSD

Reply:
Looks like I was a little to slow. Thanks @freqlabs.